tests/cli/test_side_dag.py

```diff
@@ -61,138 +55,85 @@ def test_process_argv(
-        side_dag_instance_mock.run.assert_called_once()
-        hathor_node_process.terminate.assert_called_once()
-        conn_mock.send.assert_called_once_with(SideDagProcessTerminated())
-
-
-def test_run_hathor_node_init_failed() -> None:
-    argv: list[str] = []
-    run_node_cmd_mock = Mock(side_effect=Exception)
-    capture_stdout = False
-    conn_mock = Mock()
-    parent_process_mock = Mock()
-
-    with patch('psutil.Process', return_value=parent_process_mock):
-        _run_hathor_node(argv, run_node_cmd_mock, LoggingOutput.PRETTY, capture_stdout, conn_mock)
-        run_node_cmd_mock.assert_called_once_with(argv=argv)
-        conn_mock.send.assert_called_once_with(HathorProcessInitFail(ANY))
-        parent_process_mock.terminate.assert_not_called()
-
-
-def test_run_hathor_node_side_dag_terminated() -> None:
-    argv: list[str] = []
-    run_node_instance = Mock()
-    run_node_cmd_mock = Mock(return_value=run_node_instance)
-    capture_stdout = False
-    conn_mock = Mock()
-    conn_mock.poll = Mock(return_value=True)
-    conn_mock.recv = Mock(return_value=SideDagProcessTerminated())
-    parent_process_mock = Mock()
-
-    with patch('psutil.Process', return_value=parent_process_mock):
-        _run_hathor_node(argv, run_node_cmd_mock, LoggingOutput.PRETTY, capture_stdout, conn_mock)
-        conn_mock.send.assert_called_once_with(HathorProcessInitSuccess())
-        run_node_instance.run.assert_called_once()
-        parent_process_mock.terminate.assert_not_called()
-
-
-def test_run_hathor_node_exited() -> None:
-    argv: list[str] = []
-    run_node_instance = Mock()
-    run_node_cmd_mock = Mock(return_value=run_node_instance)
-    capture_stdout = False
-    conn_mock = Mock()
-    conn_mock.poll = Mock(return_value=False)
-    parent_process_mock = Mock()
-
-    with patch('psutil.Process', return_value=parent_process_mock):
-        _run_hathor_node(argv, run_node_cmd_mock, LoggingOutput.PRETTY, capture_stdout, conn_mock)
-        conn_mock.send.assert_has_calls([call(HathorProcessInitSuccess()), call(HathorProcessTerminated())])
-        run_node_instance.run.assert_called_once()
-        parent_process_mock.terminate.assert_called_once()
-
-
-def dummy_run_hathor_node() -> None:
+def dummy_run_node_success() -> None:
+    """A dummy success is represented by a function that runs forever, like the real run_node."""
+    # We catch SIGTERM so tests make sure processes are killed
+    signal.signal(signal.SIGTERM, lambda _, __: None)
+    while True:
+        pass
+
+
+def dummy_run_node_fail() -> None:
+    """A dummy fail is represented by a function that returns immediately."""
     pass
 
 
-def test_main_success() -> None:
-    with (
-        patch.object(side_dag, '_run_side_dag_node') as run_side_dag_mock,
-        patch.object(side_dag, '_run_hathor_node', dummy_run_hathor_node),
+@patch('hathor.cli.side_dag.MONITOR_WAIT_PERIOD', 1)
+def test_both_succeed() -> None:
```

Comment:
Maybe we should use a bash script to test it because it is very hard to test this kind of behavior using unit tests.

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/1079/commits/e965a13f46986b636960070abbf4693bea9fc5c7

Reply:
Ok! Thanks!